TemperatureManager.h

```diff
@@ -0,0 +1,38 @@
+#ifndef TEMPERATUREMANAGER
+#define TEMPERATUREMANAGER
+
+class TemperatureManager : public BatteryManagementSystem
+{
+    public:
+    bool checkOptimumLimit(float temperature)
+    {
+        bool returnStatus = true;
+        if(temperature < MINIMUMTEMPERATURE || temperature > MAXIMUMTEMPERATURE)

```

Comment:
Can re-use `getRange` here, instead of repeating the comparison. When the comparison needs to change (e.g., based on type of battery), then the changes need to be done in one place.

Reply:
re-used classifyRange.